scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/BaseOxfordNanoporeProcessor.java

```diff
@@ -63,27 +68,65 @@ public PathType getPathType() {
-              // Immediately check if we're dealing with the old format, move on if so.
-              if (POSITION.matcher(path.toString()).find()) {
-                log.debug("Skipping " + path + " because we found a Sequencer Position in it.");
+              log.debug("Pre-visit: " + path);
+
+              /**
+               * If directory matches criteria we know exclude the directory from consideration,
+               * don't go into the directory (SKIP_SUBTREE)
+               */
+              if (excludedDirectoryFormat(path) || olderThanCutoff(path)) {
+                log.debug("Skipping " + path + " because we found an excluded directory in it.");
                 return FileVisitResult.SKIP_SUBTREE;
               }
+
+              /**
+               * If directory matches criteria we know include the directory, add without looking
+               * inside directory (add, SKIP_SUBTREE). Checking if result of readsDirectoryForRun
+               * 'isDirectory' is an existence check.
+               */
               if (readsDirectoryForRun(path).anyMatch(Files::isDirectory)) {
                 log.debug("Adding " + path);
                 runDirectories.add(path.toFile());
                 return FileVisitResult.SKIP_SUBTREE;
               }
-              return FileVisitResult.CONTINUE;
+
+              // If directory has directories or fast5s in, keep going through directory (CONTINUE)
+              if (Arrays.asList(path.toFile().listFiles())
+                      .stream()
+                      .filter(f -> f.isDirectory() || isFileFast5(f))
+                      .count()
+                  > 0) {
+                return FileVisitResult.CONTINUE;
+              }
+
+              // There's nothing good in this directory, so don't go within (SKIP_SUBTREE)
+              return FileVisitResult.SKIP_SUBTREE;
             }
 
+            /** Actions to take on file(s) within directory. */
             @Override
-            public FileVisitResult visitFile(Path path, BasicFileAttributes basicFileAttributes)
-                throws IOException {
-              log.debug("Visiting " + path + " and moving on");
-              return FileVisitResult.CONTINUE;
+            public FileVisitResult visitFile(Path path, BasicFileAttributes basicFileAttributes) {
+              // If file is a fast5 file, add directory path to runDirectories
+              if (isFileFast5(path)) {
+                runDirectories.add(path.getParent().toFile());
+                return FileVisitResult.SKIP_SIBLINGS;
+              } else {
+                /**
+                 * If file isn't fast5, log because it probably shouldn't be in sequencer output,
+                 * but keep going (CONTINUE) because we know from preVisitDirectory that there's
+                 * something worth looking at in here
+                 */
+                mysteryFiles.debug(path.toString());
```

Comment:
Are these files doing any harm? Are we expected to monitor this log and take some action if there are files in it?

Reply:
A large number of non-fast5 siblings to fast5 can slow down scanning, and is not expected in the sequencing output.